### PR TITLE
treesit.el Support

### PR DESCRIPTION
**Pull request by DevelopmentCool2449**
This adds real treesit.el support, also i've added a feature for unfold by clicking fold-replacement.

Some changes to consider:
* I've added some `-ts-mode` modes to `treesit-fold-range-alist`
   There are some modes that still need to add such as `vimscript-ts-mode` and more
   that i think it can be added in separated PR.
* some functions such as `treesit-fold-2str` `treesit-fold-get-children` and `treesit-fold-get-children-traverse` were replaced or removed (`treesit-fold-get-children-traverse` was unused so i commented out it).
* Most `tsc-*` functions were replaced by their `treesit` counterpart.

I've tested it with lua, c/c++ and python and should work for the other languages.
Also i've added some temporal comments for review.

### CODE REVIEW

**Comment by jcs090218** [comment]
Can you solve the compile warnings? Thanks!

```
treesit-fold.el:287:2: Warning: docstring wider than 80 characters

In end of data:
treesit-fold.el:311:28: Warning: the function ‘treesit-fold--trigger’ is not known to be defined.

Compiling /home/runner/work/treesit-fold/treesit-fold/treesit-fold-indicators.el... done ✓

In global-treesit-fold-indicators-mode:
treesit-fold-indicators.el:151:2: Warning: docstring wider than 80 characters
```

Can you also update the Eask-file? Thank you!

**Comment by DevelopmentCool2449** [comment]
> Can you solve the compile warnings? Thanks!
> 
> ```
> treesit-fold.el:287:2: Warning: docstring wider than 80 characters
> 
> In end of data:
> treesit-fold.el:311:28: Warning: the function ‘treesit-fold--trigger’ is not known to be defined.
> 
> Compiling /home/runner/work/treesit-fold/treesit-fold/treesit-fold-indicators.el... done ✓
> 
> In global-treesit-fold-indicators-mode:
> treesit-fold-indicators.el:151:2: Warning: docstring wider than 80 characters
> ```
> 
> Can you also update the Eask-file? Thank you!

Done. About the docstring warning i'm not sure why it happens, i think it can be a false-positive.

File: treesit-fold.el [hunk 1]
```diff
@@ -9,7 +9,7 @@
 ;;         Shen, Jen-Chieh <jcs090218@gmail.com>
 ;; URL: https://github.com/emacs-tree-sitter/treesit-fold
 ;; Version: 0.1.0
-;; Package-Requires: ((emacs "29.1") (tree-sitter "0.15.1") (s "1.9.0") (fringe-helper "1.0.1"))
+;; Package-Requires: ((emacs "29.1") treesit (s "1.9.0") (fringe-helper "1.0.1"))
```
Comment (jcs090218) [inline treesit-fold.el:12]: We can remove the treesit dependency here.

Comment (DevelopmentCool2449) [reply treesit-fold.el]: Sure, i've removed it.

**Comment by DevelopmentCool2449** [comment]
As an aditional note i'm wondering if we can add a section in the README file for how allow folding non-ts modes by using their parsers, it's based in a @abougouffa comment, i've tested it for elisp evaling this code snippet and folding works well.
```elisp
(when (and (treesit-available-p) (treesit-language-available-p 'elisp))
  (treesit-parser-create 'elisp))
```
or add an option for allow that.

**Comment by jcs090218** [comment]
> As an aditional note i'm wondering if we can add a section in the README file for how allow folding non-ts modes by using their parsers, it's based in a @abougouffa comment, i've tested it for elisp evaling this code snippet and folding works well.

I agreed. Let's add it to the README! :D

**Comment by DevelopmentCool2449** [comment]
Done

**Comment by jcs090218** [comment]
LGTM! Thank you so much! :D

**Review by jcs090218 — APPROVED** [review]

**Comment by abougouffa** [comment]
Excellent work @DevelopmentCool2449 !

**Pull request merged**